plugins/hydra_rq_launcher/hydra_plugins/hydra_rq_launcher/_core.py

```diff
@@ -0,0 +1,166 @@
+        job_dir_key="hydra.sweep.dir",
+        job_subdir_key="hydra.sweep.subdir",
+    )
+
+    return ret
+
+
+def launch(
+    launcher: RQLauncher, job_overrides: Sequence[Sequence[str]], initial_job_idx: int,
+) -> JobReturn:
+    """
+    :param job_overrides: a List of List<String>, where each inner list is the arguments for one job run.
+    :param initial_job_idx: Initial job idx in batch.
+    :return: an array of return values from run_job with indexes corresponding to the input list indexes.
+    """
+    setup_globals()
+    assert launcher.config is not None
+    assert launcher.config_loader is not None
+    assert launcher.task_function is not None
+
+    configure_log(launcher.config.hydra.hydra_logging, launcher.config.hydra.verbose)
+    sweep_dir = Path(str(launcher.config.hydra.sweep.dir))
+    sweep_dir.mkdir(parents=True, exist_ok=True)
+
+    # RQ configuration
+    rq_cfg = launcher.rq
+
+    # Redis configuration
+    redis_cfg = launcher.rq["redis"]
+    is_async = True
+    try:
+        is_async = not redis_cfg["mock"]
+    except ValidationError:
+        pass
+    if is_async:
+        connection = Redis(
+            host=redis_cfg["host"],
+            port=redis_cfg["port"],
+            db=redis_cfg["db"],
+            password=redis_cfg["password"],
+        )
+    else:
+        log.info("Running in synchronous mode")
+        connection = FakeStrictRedis()
+    queue = Queue(
+        name=rq_cfg["queue"],
+        connection=connection,
+        is_async=is_async,
+        serializer=cloudpickle,
+    )
+
+    # Enqueue jobs
+    jobs = []
+    singleton_state = Singleton.get_state()
+    log.info(
+        "RQ Launcher ({}) is enqueuing {} jobs".format(
+            ",".join([f"{k}={v}" for k, v in rq_cfg.items()]), len(job_overrides),
+        )
+    )
+    log.info("Enqueueing jobs, sweep output dir : {}".format(sweep_dir))
+    for idx, overrides in enumerate(job_overrides):
+        description = " ".join(filter_overrides(overrides))
+        log.info(f"\t#{idx} : {description}")
+
+        enqueue_keywords = OmegaConf.to_container(rq_cfg["enqueue"], resolve=True)
+        if enqueue_keywords["job_id"] is None:
+            enqueue_keywords["job_id"] = str(uuid.uuid4())
+        if enqueue_keywords["description"] is None:
+            enqueue_keywords["description"] = description
+
+        job = queue.enqueue(
+            execute_job,
+            idx=initial_job_idx + idx,
+            overrides=overrides,
+            config_loader=launcher.config_loader,
+            config=launcher.config,
+            task_function=launcher.task_function,
+            singleton_state=singleton_state,
+            redis_job_id=enqueue_keywords["job_id"],
+            **enqueue_keywords,
+        )
+        jobs.append(job)
+
+        log.info(f"Enqueued {job.get_id()}")
+
+    log.info("Finished enqueuing")
+    if rq_cfg["stop_after_enqueue"]:
+        raise StopAfterEnqueue
```

Comment:
Can you show me the console output when this happens?
what does it look like to the user?

Reply:
Sure, this is how it would look like:
```
[2020-06-19 16:20:52,189][HYDRA] RQ Launcher is enqueuing 3 job(s) in queue : default
[2020-06-19 16:20:52,190][HYDRA] Sweep output dir : multirun/2020-06-19/16-20-52
[2020-06-19 16:20:52,190][HYDRA] 	#1 : task=1
[2020-06-19 16:20:52,216][HYDRA] Enqueued 3ae6c637-529a-43cd-8664-6ce419d2ace4
[2020-06-19 16:20:52,216][HYDRA] 	#2 : task=2
[2020-06-19 16:20:52,240][HYDRA] Enqueued 50408b07-51f4-42ce-b5c7-f586d17f9d61
[2020-06-19 16:20:52,240][HYDRA] 	#3 : task=3
[2020-06-19 16:20:52,263][HYDRA] Enqueued e7118216-aff5-479f-bf1d-9f34f47807d0
[2020-06-19 16:20:52,264][HYDRA] Finished enqueuing
Traceback (most recent call last):
  File "example/my_app.py", line 19, in <module>
    my_app()
  File "/Users/jm/GitHub/jan-matthis/hydra/hydra/main.py", line 37, in decorated_main
    strict=strict,
  File "/Users/jm/GitHub/jan-matthis/hydra/hydra/_internal/utils.py", line 270, in run_hydra
    lambda: hydra.multirun(
  File "/Users/jm/GitHub/jan-matthis/hydra/hydra/_internal/utils.py", line 188, in run_and_report
    return func()
  File "/Users/jm/GitHub/jan-matthis/hydra/hydra/_internal/utils.py", line 273, in <lambda>
    overrides=args.overrides,
  File "/Users/jm/GitHub/jan-matthis/hydra/hydra/_internal/hydra.py", line 127, in multirun
    return sweeper.sweep(arguments=task_overrides)
  File "/Users/jm/GitHub/jan-matthis/hydra/hydra/_internal/core_plugins/basic_sweeper.py", line 113, in sweep
    results = self.launcher.launch(batch, initial_job_idx=initial_job_idx)
  File "/Users/jm/GitHub/jan-matthis/hydra/plugins/hydra_rq_launcher/hydra_plugins/hydra_rq_launcher/rq_launcher.py", line 43, in launch
    launcher=self, job_overrides=job_overrides, initial_job_idx=initial_job_idx
  File "/Users/jm/GitHub/jan-matthis/hydra/plugins/hydra_rq_launcher/hydra_plugins/hydra_rq_launcher/_core.py", line 133, in launch
    raise StopAfterEnqueue
hydra_plugins.hydra_rq_launcher._core.StopAfterEnqueue
```

Reply:
FR is open here: https://github.com/facebookresearch/hydra/issues/692